packages/hardhat-plugin/src/ignition-helper.ts

```diff
@@ -55,52 +81,123 @@ export class IgnitionHelper {
       provider: this._provider,
       deploymentDir: this._deploymentDir,
       artifactResolver,
-      ignitionModule: ignitionModuleDefinition,
+      ignitionModule,
       deploymentParameters: parameters,
       accounts,
       verbose: false,
     });
 
-    if (result.status === "timeout") {
+    if (result.type === DeploymentResultType.VALIDATION_ERROR) {
+      const errorsList = Object.entries(result.errors).flatMap(
+        ([futureId, errors]) => errors.map((err) => `  * ${futureId}: ${err}`)
+      );
+
       throw new IgnitionError(
-        `The deployment has been halted due to transaction timeouts:\n  ${result.timeouts
-          .map((t) => `${t.txHash} (${t.futureId}/${t.executionId})`)
-          .join("\n  ")}`
+        `The deployment wasn't run because of the following validation errors:
+
+${errorsList.join("\n")}`
       );
     }
 
-    if (result.status !== "success") {
-      // TODO: Show more information about why it failed
-      throw new IgnitionError("Failed deployment");
-    }
+    if (result.type === DeploymentResultType.RECONCILIATION_ERROR) {
+      const errorsList = Object.entries(result.errors).flatMap(
+        ([futureId, errors]) => errors.map((err) => `  * ${futureId}: ${err}`)
+      );
 
-    return this._toEthersContracts(result);
-  }
+      throw new IgnitionError(
+        `The deployment wasn't run because of the following reconciliation errors:
 
-  private async _toEthersContracts(
-    result: DeploymentResultSuccess
-  ): Promise<Record<string, Contract>> {
-    const resolvedOutput: { [k: string]: Contract } = {};
+${errorsList.join("\n")}`
+      );
+    }
 
-    for (const [key, future] of Object.entries(result.module.results)) {
-      const deployedContract = result.contracts[future.id];
+    if (result.type === DeploymentResultType.EXECUTION_ERROR) {
+      console.log(result);
+      throw new IgnitionError(
+        `The deployment wan't succesful: TODO: create a good error message`
```

Comment:
This is also something else that you can make progress on, @kanej

Reply:
What do you think of only showing the failed and timed out within the Ignition Error:

```
The deployment wan't succesful, there were timeouts and errors:

Timed out
  * SomeContract

Errors
  * MyContract: reverted with reason z
  * AnotherContract: reverted with reason y
```